osu.Game/Rulesets/RulesetConfigCache.cs

```diff
@@ -15,12 +16,31 @@ namespace osu.Game.Rulesets
     /// </summary>
     public class RulesetConfigCache : Component
     {
-        private readonly ConcurrentDictionary<int, IRulesetConfigManager> configCache = new ConcurrentDictionary<int, IRulesetConfigManager>();
-        private readonly SettingsStore settingsStore;
+        private readonly RealmContextFactory realmFactory;
+        private readonly RulesetStore rulesets;
 
-        public RulesetConfigCache(SettingsStore settingsStore)
+        private readonly Dictionary<int, IRulesetConfigManager> configCache = new Dictionary<int, IRulesetConfigManager>();
```

Comment:
I'm a little bit suspicious of this change from `ConcurrentDictionary`, given that the original change was done with intent: https://github.com/ppy/osu/pull/4730

Granted, it's a one-frame possibility. Or maybe not because of `Dispose()` which could be async?

Reply:
Everything is inserted immediately now, on startup, so it shouldn’t be an issue I think.